src/odemis/gui/cont/settings.py

```diff
@@ -637,8 +628,8 @@ def on_acqfile_change(self, file_info):
                 self._pnl_acqfile.add_metadata(key, value)
 
             # Change default dir for the calibration files
-            for file_ctrl in (self._arfile_ctrl, self._spec_bckfile_ctrl,
-                              self._temporalspec_bckfile_ctrl, self._specfile_ctrl):
+            for file_ctrl in (self._ar_bckfile_entry.value_ctrl, self._spec_bckfile_entry.value_ctrl,
+                              self._temporalspec_bckfile_entry.value_ctrl, self._specfile_entry.value_ctrl):
                 file_ctrl.default_dir = file_info.file_path
```

Comment:
The point of this loop is to avoid repeating the code, so you could write more concisely:
```
            for file_entry in (self._ar_bckfile_entry., self._spec_bckfile_entry,
                              self._temporalspec_bckfile_entry, self._specfile_entry):
                file_entry.value_ctrl.default_dir = file_info.file_path
```